### PR TITLE
[misc] Fix outdated links

### PR DESCRIPTION
**Pull request by khluu**
## Why are these changes needed?
Outdated links are causing Lint test to fail on `postmerge` CI pipeline.

## Changes
- Links updated with new ones that I manually verify to be valid.
- Added `madewithml.com/#course` to the ignore list for linkcheck. My guess is that the host for this domain doesn't accept Sphinx user agent.

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
use `-s` on the commits so that it can pass the `DCO` test?

File: doc/source/conf.py [hunk 1]
```diff
@@ -238,6 +238,7 @@ def __init__(self, version: str):
         r"http://localhost[:/].*",  # Ignore localhost links
         r"^http:/$",  # Ignore incomplete links
         # 403 Client Error: Forbidden for url.
+        "https://madewithml.com/#course",
```
Comment (aslonnie) [inline doc/source/conf.py:241]: why is this one added here?

Comment (khluu) [reply doc/source/conf.py:241]: I suspect that the host server for `madewithml.com` somehow doesn't accept Sphinx user agent for linkcheck, thus the 403 error. I manually verify and the link seems to work, so I added this domain to the ignore list.

**Comment by aslonnie** [comment]
could you fix the reviewer list?

File: doc/source/ray-core/examples/overview.rst [hunk 2]
```diff
@@ -55,7 +55,7 @@ Reinforcement Learning Examples
 -------------------------------
 
 These are simple examples that show you how to leverage Ray Core.
-For Ray's production-grade reinforcement learning library, see `RLlib <http://docs.ray.io/en/latest/rllib.html>`__.
+For Ray's production-grade reinforcement learning library, see `RLlib <https://docs.ray.io/en/latest/rllib/index.html>`__.
```
Comment (aslonnie) [inline doc/source/ray-core/examples/overview.rst:58]: I think each of these files require different reviewers as they are in different directories. could you separate each file into different PRs, so that they only require a subset of people to approve, but not all of them at once?

**Comment by can-anyscale** [comment]
w00t thank you, CC: @angelinalg will love this

File: doc/source/ray-core/examples/overview.rst (hunk 2 above)
Comment (khluu) [reply doc/source/ray-core/examples/overview.rst:58]: Ya that's a great idea! The list of reviewers in this PR keeps going on and on.
